modules/volunteers/admin/sitesupport/giftTable.js

```diff
@@ -0,0 +1,297 @@
+      .then((response) => {
+        const result = JSON.parse(response.responseText);
+        this.records = result.data;
+
+        this.records.sort((a,b) => {
+          if (a.value == b.value) {
+            return ((a.name > b.name) ? 1 : -1);
+          }
+          return ((parseFloat(a.value) > parseFloat(b.value)) ? 1 : -1);
+        });
+
+        this.records.sort((a,b) => {
+          if (a.reward_group != null && b.reward_group == null) {return -1;}
+          if (b.reward_group != null && a.reward_group == null) {return 1;}
+          if (b.reward_group == null && a.reward_group == null) {return 0;}
+          if (parseInt(b.reward_group.id) == parseInt(a.reward_group.id)) {
+            if (a.value == b.value) {
+              return ((a.name > b.name) ? 1 : -1);
+            }
+            return (parseFloat(a.value) > parseFloat(b.value)) ? 1 : -1;
+          }
+          return (parseInt(a.reward_group.id) > parseInt(b.reward_group.id) ? 1 : -1);
+        });
+
+        if (userId != null) {
+          apiRequest('GET', '/member/' + userId + '/volunteer/claims', 'max_results=all')
+            .then((response) => {
+              const result = JSON.parse(response.responseText);
+              this.claims = result.data;
+              result.data.forEach((entry) => {
+                var rewardId = entry.reward.id;
+                this.records.every((record) => {
+                  if (record.id == rewardId) {
+                    if ('acquired' in record) {
+                      record.acquired += 1;
+                    } else {
+                      record['acquired'] = 1;
+                    }
+                    if (record.reward_group != null) {
+                      if (record.reward_group.id in this.groupCount) {
+                        this.groupCount[record.reward_group.id] += 1;
+                      } else {
+                        this.groupCount[record.reward_group.id] = 1;
+                      }
+                    }
+                    return false;
+                  }
+                  return true;
+                });
+              });
+              this.loaded = true;
+            })
+            .catch(() => {
+              this.loaded = true;
+            });
+        } else {
+          this.loaded = true;
+        }
+      })
+      .catch((e) => {
+        console.log(e);
+      });
+  },
+  created() {
+  },
+  data() {
+    return {
+      title: 'Gifts',
+      columns: null,
+      records: [],
+      claims: [],
+      hideSoldOut: true,
+      hoursRemaining: 0,
+      unclaimed: null,
+      groupCount: [],
+      loaded: false,
+    }
+  },
+  beforeUpdate() {
+    this.groupIndex = -1;
+    this.groupColorId = null;
+    this.lastGroup = null;
+    this.hoursRemaining = this.totalHours - this.hoursSpent;
+  },
+  methods: {
+    clicked(record) {
+      if (this.$parent.userId == null && this.$parent.isAdmin) {
```

Comment:
What is `$parent` here? If we're using this to refer to other components, could we take advantage of component registration instead?

https://vuejs.org/guide/components/registration.html#local-registration

Reply:
the $parent is the parent component in the DOM.  in this case it is `modules/volunteers/admin/sitesupport/vue.js` that holds all the components of the admin page.